org.mwc.cmap.NarrativeViewer/src/org/mwc/cmap/NarrativeViewer2/NATViewerView.java

```diff
@@ -817,20 +818,25 @@ public void run() {
 			public void runWithEvent(Event event) {
 				//show file dialog
 				//default directory, last directory, otherwise the user's home folder.
-				DirectoryDialog dialog = new DirectoryDialog(getViewSite().getShell());
-				String prefDir = Activator.getInstance().getPreferenceStore().getString(PREF_EXPORT_DIR);
-				String defaultDir = System.getProperty("user.home");
-				if(prefDir == null || prefDir.isBlank()) {
-					dialog.setFilterPath(defaultDir);
+				if(myViewer.getVisibleNarratives()!=null && myViewer.getVisibleNarratives().size()>0) {
+					DirectoryDialog dialog = new DirectoryDialog(getViewSite().getShell());
+					String prefDir = Activator.getInstance().getPreferenceStore().getString(PREF_EXPORT_DIR);
+					String defaultDir = System.getProperty("user.home");
+					if(prefDir == null || prefDir.isBlank()) {
+						dialog.setFilterPath(defaultDir);
+					}
+					else {
+						dialog.setFilterPath(prefDir);
+					}
+					String targetDirectory = dialog.open();
+					if(targetDirectory!=null) {
+						Activator.getInstance().getPreferenceStore().setValue(PREF_EXPORT_DIR, targetDirectory);
+						NarrativeWriter writer = new NarrativeWriter();
+						writer.write(myViewer.getVisibleNarratives(), myViewer.isShowingSource(), myViewer.isShowingType(), new File(targetDirectory));
+					}
 				}
 				else {
-					dialog.setFilterPath(prefDir);
-				}
-				String targetDirectory = dialog.open();
-				if(targetDirectory!=null) {
-					Activator.getInstance().getPreferenceStore().setValue(PREF_EXPORT_DIR, targetDirectory);
-					NarrativeWriter writer = new NarrativeWriter();
-					writer.write(myViewer.getVisibleNarratives(), myViewer.isShowingSource(), myViewer.isShowingType(), new File(targetDirectory));
+					MessageDialogWithToggle.openWarning(getViewSite().getShell(), "Error exporting narratives", "No narratives to export");
```

Comment:
@ayeshare - is this the correct dialog class?  We don't have anything to toggle.

Reply:
@IanMayo thats right. Changed it to MessageDialog.